homeassistant/components/lyric/__init__.py

```diff
@@ -0,0 +1,197 @@
+        LyricLocalOAuth2Implementation(
+            hass,
+            DOMAIN,
+            config[DOMAIN][CONF_CLIENT_ID],
+            config[DOMAIN][CONF_CLIENT_SECRET],
+            OAUTH2_AUTHORIZE,
+            OAUTH2_TOKEN,
+        ),
+    )
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Honeywell Lyric from a config entry."""
+    implementation = (
+        await config_entry_oauth2_flow.async_get_config_entry_implementation(
+            hass, entry
+        )
+    )
+
+    session = aiohttp_client.async_get_clientsession(hass)
+    oauth_session = config_entry_oauth2_flow.OAuth2Session(hass, entry, implementation)
+
+    client = ConfigEntryLyricClient(session, oauth_session)
+
+    client_id = hass.data[DOMAIN][CONF_CLIENT_ID]
+    lyric: Lyric = Lyric(client, client_id)
+
+    try:
+        await lyric.get_locations()
+    except LYRIC_EXCEPTIONS as exception:
+        _LOGGER.warning(exception)
+        raise ConfigEntryNotReady from exception
+
+    async def async_update_data() -> Lyric:
+        """Fetch data from Lyric."""
+        async with async_timeout.timeout(60):
+            try:
+                await lyric.get_locations()
+            except LYRIC_EXCEPTIONS as exception:
+                raise UpdateFailed(exception) from exception
+            return lyric
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        # Name of the data. For logging purposes.
+        name="lyric_coordinator",
+        update_method=async_update_data,
+        # Polling interval. Will only be polled if there are subscribers.
+        update_interval=timedelta(seconds=120),
+    )
+
+    hass.data[DOMAIN][entry.entry_id] = coordinator
+
+    # Fetch initial data so we have data when entities subscribe
+    await coordinator.async_refresh()
+
+    for component in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, component)
+        )
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Unload a config entry."""
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(entry, component)
+                for component in PLATFORMS
+            ]
+        )
+    )
+    if unload_ok:
+        hass.data[DOMAIN].pop(entry.entry_id)
+
+    return unload_ok
+
+
+class LyricEntity(CoordinatorEntity):
+    """Defines a base Honeywell Lyric entity."""
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        location: LyricLocation,
+        device: LyricDevice,
+        key: str,
+        name: str,
+        icon: str,
+    ) -> None:
+        """Initialize the Honeywell Lyric entity."""
+        super().__init__(coordinator)
+        self._update_thermostat = coordinator.data.update_thermostat
+        self._device = device
```

Comment:
It looks like you only need `self._device.macID`.

```suggestion
        self._macid = device.macID
```